### PR TITLE
Add color to commands output

### PR DESCRIPTION
**Pull request by undef1nd**

### CODE REVIEW

**Comment by undef1nd** [comment]
@arusahni Hey, I added colors. I had to merge stdout & stderr output into single one in order to make messages from git colorful. Please, take a look. Thanks.

**Comment by arusahni** [comment]
@undef1nd this is a great start!

I poked around a bit at `duct`, and think I found a way to keep stderr and stdout separated for times it may be necessary. Specifically,

```rust
let output_obj = cmd!("foo"),stderr_capture().stdout_capture().run();
let output = match output_obj {
   Ok(obj) => String::from_utf8(obj.stdout.to_vec()).unwrap(),
   Err(obj) => String::from_utf8(obj.stderr.to_vec()).unwrap(),
};
```
The downside of this approach is it doesn't contain the time lines were written to the buffer, making it tough to interleave the two in a reasonable way.

*So*, all of this is a long way for me to say: let's keep the current merging technique.  For coloring, lets follow this set of rules:

* Informational text (e.g., `git fetch` output): uncolored
* Success text (e.g., "Domain key changed!"): green
* Error text (e.g., "There was an error checking out the branch"): red

Thanks!

**Comment by undef1nd** [comment]
@arusahni ,
Thanks for sharing your finding! I was not able to figure out it myself.

Let me clarify reg the information text. If we want to keep it uncolored, I guess, I can remove merging stdout and stderr I added in my previous common. Am I right?

**Comment by undef1nd** [comment]
In the meantime I've changed colors to green and red.

**Comment by arusahni** [comment]
> Let me clarify reg the information text. If we want to keep it uncolored, I guess, I should just remove merging stdout and stderr I added in my previous commit. Should I?

Yeah. Pretty much anything decorated with `blue()`, I think.

Also, sorry about the build errors - it's not your fault. I've been experimenting with CI changes on `master`, forgetting that others are actively working on the codebase. I should be resolving that tonight.

**Comment by undef1nd** [comment]
Hey @arusahni , 
I've made the changes. Plz, take a look. Let me know if you have any concerns.

File: src/remotes/mod.rs [hunk 1]
```diff
@@ -65,8 +66,13 @@ fn get_api_key(domain: &str) -> String {
...
         None => {
             let mut newkey = String::new();
-            println!("No API token for {} found. See https://github.com/arusahni/git-req/wiki/API-Keys for instructions.", domain);
-            print!("{} API token: ", domain);
+            println!(
+                "{}",
+                format!("No API token for {} found. See https://github.com/arusahni/git-req/wiki/API-Keys for instructions.", domain).red()
+                );
+            print!("{}",
+                   format!("{} API token: ", domain).red()
```
Comment (arusahni) [inline src/remotes/mod.rs]: Since this is a prompt, let's leave it uncolored.

File: src/main.rs [hunk 2]
```diff
@@ -119,9 +125,15 @@ fn list_open_requests() {
     let mut tw = TabWriter::new(io::stdout()).padding(4);
     for mr in &mrs {
         if remote.has_useful_branch_names() {
-            writeln!(&mut tw, "{}\t{}\t{}", mr.id, mr.source_branch, mr.title).unwrap();
+            writeln!(
+                &mut tw,
+                "{}", format!("{}\t{}\t{}", mr.id, mr.source_branch, mr.title).green()
```
Comment (arusahni) [inline src/main.rs]: It'd be easier to visually parse this output if we apply different colors. I'm thinking that: `mr.id` is the most important, so it should stand out.  Maybe `green` for the `mr.id`, `dimmed` for the `source_branch` and uncolored for the `title`?  I can worry more about tweaking these in future versions.

File: src/main.rs [hunk 3]
```diff
@@ -119,9 +125,15 @@ fn list_open_requests() {
...
+            ).unwrap();
         } else {
-            writeln!(&mut tw, "{}\t{}", mr.id, mr.title).unwrap();
+            writeln!(
+                &mut tw,
+                "{}", format!("{}\t{}", mr.id, mr.title).green()
```
Comment (arusahni) [inline src/main.rs]: Same color guidance as above: `green` and `uncolored`.

**Comment by arusahni** [comment]
Definitely along the lines of what I was thinking. If you address the comments above, content-wise I think we're good to go.

Also, before we merge, can you run [`rustmft`](https://github.com/rust-lang/rustfmt) and [`clippy`](https://github.com/Manishearth/rust-clippy) against the code?

Thanks!

File: src/remotes/mod.rs (hunk 1 above)
Comment (undef1nd) [reply src/remotes/mod.rs]: I'm not quite sure here. Should I uncolor both - 71 and 74 lines or only 74?

Comment (undef1nd) [reply src/remotes/mod.rs]: You're right - both are of single prompt. Will remove color from them both to be consistent. Sorry, never mind my prev.comment 🙂

File: src/main.rs (hunk 2 above)
Comment (undef1nd) [reply src/main.rs]: Done.

File: src/main.rs (hunk 3 above)
Comment (undef1nd) [reply src/main.rs]: Done.

**Comment by undef1nd** [comment]
@arusahni Guess it’s ready for the final review. Hope I didn’t miss anything.

**Comment by arusahni** [comment]
@undef1nd this is great! Thanks for working with me to get it to this state (and for your contribution)!

**Comment by undef1nd** [comment]
@arusahni thanks a lot for your help and guidance!

**Pull request merged**